src/Sentry.Protocol/User.cs

```diff
@@ -45,6 +47,15 @@ public class User
         [DataMember(Name = "username", EmitDefaultValue = false)]
         public string Username { get; set; }
 
+        /// <summary>
+        /// Additional informations about the user
+        /// </summary>
+        /// <value>
+        /// Additional informations about the user
+        /// </value>
+        [DataMember(Name = "extras", EmitDefaultValue = false)]
+        public Dictionary<string, string> Extras { get; set; }
```

Comment:
Could you please split it in an `internal Dictionary<string, string,> InternalOther` and `public IReadOnlyDictionary Other`? Something like:

https://github.com/getsentry/sentry-dotnet-protocol/blob/master/src/Sentry.Protocol/BaseScope.cs#L39-L40
And
https://github.com/getsentry/sentry-dotnet-protocol/blob/master/src/Sentry.Protocol/BaseScope.cs#L154-L160

No reason to be thread-safe though.
So we can create a new Dictionary on the getter if required and also the public `Other` can have a public setter so that the whole collection can be replaced.